src/mem/cache/prefetch/bop.cc

```diff
@@ -201,27 +207,34 @@ BOP::bestOffsetLearning(Addr x)
         }
     }
 
+    // Move the offset iterator forward to prepare for the next time
     offsetsListIterator++;
 
     // All the offsets in the list were visited meaning that a learning
     // phase finished. Check if
     if (offsetsListIterator == offsetsList.end()) {
         offsetsListIterator = offsetsList.begin();
         round++;
+    }
 
-        // Check if the best offset must be updated if:
-        // (1) One of the scores equals SCORE_MAX
-        // (2) The number of rounds equals ROUND_MAX
-        if ((bestScore >= scoreMax) || (round == roundMax)) {
-            bestOffset = phaseBestOffset;
+    // Check if its time to re-calculate the best offset
+    if ((bestScore >= scoreMax) || (round >= roundMax))
+    {
             round = 0;
+            if (bestScore > badScore)
```

Comment:
Could you document why this is checked? It has been years since I reviewed this code/properly read the paper, so I don´t recall how it should work anymore.

Reply:
Fixed in the latest commit